### ISSUE TITLE
Feature request: Example of a more modern SPA dotnet app

### ISSUE THREAD

**Issue by gilliganc** [kinde-starter-kits/dotnet-starter-kit#10]
### Prerequisites

- [X] I have searched the repository’s issues and [Kinde community](https://thekindecommunity.slack.com/archives/C04K34HUV9B) to ensure my feature request isn’t a duplicate
- [X] I have read the [contributing guidelines](https://github.com/kinde-starter-kits/.github/blob/ff8d5c462e56772810bca53f750fd3610a00d673/.github/CONTRIBUTING.md)
- [X] I agree to the terms in the [code of conduct](https://github.com/kinde-starter-kits/.github/blob/ff8d5c462e56772810bca53f750fd3610a00d673/.github/CODE_OF_CONDUCT.md)

### What is the problem you’re trying to solve?

All the dotnet documentation i discovered even when you click on backend on your website talk about your SDK and authentication redirects etc etc. When i use other providers like Auth0 in a Rect SPA talking to dotnet lambda rest API's as the backend i can actually just use dotnets built in JWT support to verify the token easily and extract information. 

### What solution would you like to see?

I would like to see an example of a standalone dotnet rest service that gets called by an SPA using your authentication and it only implements what is needed to handle and authenticate the request correctly, so verify the JWT and authorization for method access etc. If it really needs your SDK and all that setup that is in the example that is fine but that to mean seems more like a setup when using dotnet for both frontend and backend in one not using dotnet for backend rest services like most may think of when you say backend dotnet vs listing it under frontend.

**Comment by leeleito** [comment]
Agree lack of .net moderen examples is not giving us confidence in the product. Especially when the this SDK has not been updated in sometime.  Everyone else can supply Blazor Server, Blazor WASM examples. See Logto, Auth0

**Comment by coel** [comment]
Thank you for the feedback.  We will likely publish a starter for Web API projects that can be used for reference.

To clarify, Kinde follows the OpenID Connect specification and as .NET has built-in libraries that support this it is not necessary to the use the SDK.  Where a JWT is passed to the backend typically the `JwtBearer` middleware would be used.

**Comment by leeleito** [comment]
It might be a good idea then to supply a simple example using the .NET built in libraries. Showing how to use the settings from Kinde.

**Comment by gilliganc** [comment]
Kinde support also provided me with settings needed to make the Swagger Ui work on the Web API project with authentication button which was extremely useful for my team

**Comment by gilliganc** [comment]
> To clarify, Kinde follows the OpenID Connect specification and as .NET has built-in libraries that support this it is not necessary to use the SDK. Where a JWT is passed to the backend typically the `JwtBearer` middleware would be used.

This statement is mostly right from our testing but where the dotnet built in  `JwtBearer` middleware will have issues is the format kinde uses for the roles in the JWT access token if you enable them. It is an array of objects vs a list of string like permissions are so custom code or SDK would likely be needed for that. We have ignored that for now and just use permissions.

**Comment by coel** [comment]
Just to follow-up, we have released a guide to securing web API projects with Kinde: https://docs.kinde.com/developer-tools/your-apis/dotnet-based-apis/

We have also released a starter kit for web API projects that contains a complete example: https://github.com/kinde-starter-kits/dotnet-webapi-starter-kit.

I have had a further look into handling role.  Microsoft's schema expects a `role` claim, the closest I could find in the OAuth 2.0 spec is a `roles` (with an `s`) claim, but has no canonical types.  Returning the roles in the expected format could be something we support via a token customisation in future.

ASP.NET does support creating custom policies that can inspect any claims, so provides more flexibility.  Creating a custom policy to secure endpoints based on the roles defined in Kinde is the approach we have taken in the guide and starter kit.
